packages/proto-loader/src/index.ts

```diff
@@ -368,6 +379,38 @@ export function loadSync(
   return createPackageDefinition(root, options!);
 }
 
+export async function loadFileDescriptorSet(
+  descriptorSet: Protobuf.Message<descriptor.IFileDescriptorSet> &
```

Comment:
Instead of a `Message`, the input here should be a `Buffer` containing the binary-encoded message, which should then be decoded within the function. Alternatively, or in addition, the input could be a plain JavaScript object that could be passed to `Protobuf.Message<descriptor.IFileDescriptorSet>.fromObject()`.

As I mentioned in the issue, hiding the Protobuf.js API is a core design goal of this library. That means that the user should not need to have any interaction with the Protobuf.js library directly in order to use any of the APIs exported from this library.

Reply:
That makes sense. Thanks for the advice.